File: src/test/php/DateTimeUtilsTest.php
<?php

declare(strict_types=1);

namespace PetrKnap\Persistence\ZonedDateTime;

use DateInterval;

final class DateTimeUtilsTest extends TestCase
{
    public function test_asUtcInstantAtOffset(): void
    {
        self::assertEquals(
            $this->utcDateTime->add(new DateInterval('PT' . self::ZONED_DATETIME_OFFSET . 'S'))->getTimestamp(),
            DateTimeUtils::asUtcInstantAtOffset($this->localDateTime, self::ZONED_DATETIME_OFFSET)->getTimestamp(),
        );
    }

    public function test_parseAsLocalDateTime(): void
    {
        self::assertDateTimeEquals(
            $this->localDateTime,
            DateTimeUtils::parseAsLocalDateTime(self::LOCAL_DATETIME, self::LOCAL_DATETIME_FORMAT),
        );
    }

    public function test_parseAsLocalDateTime_throws_on_incorrect_datetime(): void
    {
        self::expectException(Exception\DateTimeUtilsCouldNotParseAsLocalDateTime::class);

        DateTimeUtils::parseAsLocalDateTime('this is not a local date-time', self::LOCAL_DATETIME_FORMAT);
    }

    public function test_secondsBetween(): void
    {
        self::assertEquals(
            self::ZONED_DATETIME_OFFSET,
            DateTimeUtils::secondsBetween(JavaSe8\Time::toLocalDateTime($this->utcDateTime), $this->localDateTime),
        );
    }
}

File: src/test/java/io/github/petrknap/persistence/zoneddatetime/DateTimeUtilsTest.java
package io.github.petrknap.persistence.zoneddatetime;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

final class DateTimeUtilsTest extends TestCase {
    @Test void asUtcInstantAtOffset() {
        assertEquals(
                utcDateTime.plusSeconds(ZONED_DATETIME_OFFSET).toInstant(),
                DateTimeUtils.asUtcInstantAtOffset(localDateTime, ZONED_DATETIME_OFFSET).toInstant()
        );
    }

    @Test void parseAsLocalDateTime() {
        assertEquals(localDateTime, DateTimeUtils.parseAsLocalDateTime(LOCAL_DATETIME, LOCAL_DATETIME_FORMAT));
    }

    @Test void parseAsLocalDateTime_throws_on_incorrect_datetime() {
        assertThrows(
                DateTimeUtils.Exception.CouldNotParseAsLocalDateTime.class,
                () -> DateTimeUtils.parseAsLocalDateTime("this is not a local date-time", LOCAL_DATETIME_FORMAT)
        );
    }

    @Test void secondsBetween() {
        assertEquals(ZONED_DATETIME_OFFSET, DateTimeUtils.secondsBetween(utcDateTime.toLocalDateTime(), localDateTime));
    }
}
